apps/portal/src/app/views/notifications/notifications.component.ts

```diff
@@ -38,26 +53,40 @@ export class NotificationsComponent implements OnInit {
 
   jsonDialogVisible: Boolean = false;
 
-  deliveryStatusMap = {
-    [DeliveryStatus.PENDING]: { value: 'Pending', style: 'pending' },
-    [DeliveryStatus.IN_PROGRESS]: { value: 'In Progress', style: 'in-progress' },
-    [DeliveryStatus.SUCCESS]: { value: 'Success', style: 'success' },
-    [DeliveryStatus.FAILED]: { value: 'Failed', style: 'failed' },
-  };
-
-  channelTypeMap = ChannelTypeMap;
-
   constructor(private notificationService: NotificationsService) {}
 
   ngOnInit(): void {
     this.loadNotifications();
   }
 
   loadNotifications() {
-    this.notificationService.getNotifications().subscribe(
+    const variables = { filters: [] };
+
+    if (this.selectedChannelType) {
+      variables.filters.push({
+        field: 'channelType',
+        operator: 'eq',
+        value: this.selectedChannelType.toString(),
+      });
+    }
+
+    if (this.selectedDeliveryStatus) {
+      variables.filters.push({
+        field: 'deliveryStatus',
+        operator: 'eq',
+        value: this.selectedDeliveryStatus.toString(),
+      });
+    }
+
+    this.notificationService.getNotifications(variables).subscribe(
       (notifications: Notification[]) => {
         this.notifications = notifications;
         this.applyFilters();
+        /**
```

Comment:
See if this conversation has any useful information regarding your issue.
https://github.com/primefaces/primeng/issues/11898#issuecomment-1831076132

Reply:
Upon merging code from main branch with this branch in https://github.com/OsmosysSoftware/osmo-x/pull/165/commits/3f890a30e2c547386989eb8f7baf725258d4d24d, it was found that manually setting the `first` property of table was giving console errors. 
Upon checking, it was noticed we no longer need to manually set this property for the bug with page overflow anymore. It is possible this was resolved by the loading implementation which hides and un-hides the table. Fixed in https://github.com/OsmosysSoftware/osmo-x/pull/165/commits/47b00b8a0f3b2bcb1d6ee717c79f09e0fe51d27d